apps/ejabberd/src/ejabberd_sm.erl

```diff
@@ -293,31 +293,35 @@ get_raw_sessions(User, Server) ->
     clean_session_list(
       ?SM_BACKEND:get_sessions(jid:nodeprep(User), jid:nameprep(Server))).
 
--spec set_presence(SID, User, Server, Resource, Prio, Presence, Info) -> ok when
+-spec set_presence(Acc, SID, User, Server, Resource, Prio, Presence, Info) -> Acc1 when
```

Comment:
The same as in other places: set_presence calls `set_presence_hook`, and itself is called by c2s:update_priority, which is called by presence_update, which is called by process_outgoing_stanza, so they are all parts of the main stanza processing chain, and as such should receive an accumulator and be given a change to modify it.